### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by thyal**
Convert lib to Typescript. I've added types for a bunch things, but there's still a bunch of `any` and missing stuff. This needs to be updated incrementally. 

* All files converted to TS
* Added types for a bunch of core functionality
* Added build step with rollup
* Changed eslint config to align with other repos.

This should be tested quite a lot before merging. Test with PWA PR https://github.com/whereby/pwa/pull/3989

<details>
  <summary>📦 Published PR as canary version: <code>2.0.1--canary.88.8373353702.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @whereby/jslib-media@2.0.1--canary.88.8373353702.0
  # or 
  yarn add @whereby/jslib-media@2.0.1--canary.88.8373353702.0
  ```
</details>

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+        "ecmaFeatures": {
+            "experimentalObjectRestSpread": true,
+        },
     },
-    "plugins": ["prettier", "jest"],
+    "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
+    "plugins": ["jest"],
     "rules": {
         "no-console": ["error", { "allow": ["warn", "error"] }],
-        "prettier/prettier": "error"
```
Comment (sebapawlak) [inline .eslintrc:17]: Why did you remove this?

**Review by sebapawlak — COMMENTED** [review]
Haven't tested yet. There is a lot of `any`s and `ts-ignore`s but I guess we could live with that and iterate on them. Perhaps we should have the whole team do like a 30 minute smoke testing session of this?

File: .eslintrc (hunk 1 above)
Comment (thyal) [reply .eslintrc:17]: Copied the eslint config from one of our other Typescript projects. But I can add this back

**Comment by thyal** [comment]
> Haven't tested yet. There is a lot of `any`s and `ts-ignore`s but I guess we could live with that and iterate on them. 

I think that's what we have to do. It was too much for me to go through all of them in this PR. But the goal should be to replace all of the `any`s and `ts-ignore`s. We can create follow-up tickets for that once this is merged.

File: src/utils/types.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+/* 
+    Socket
+*/
+
+export interface SocketConf {
+    host?: string;
+    path?: string;
+    reconnectionDelay?: number;
+    reconnectoinDelayMax?: number;
```
Comment (pnts-se-whereby) [inline src/utils/types.ts:17]: probably typo

File: src/utils/types.ts
Comment (pnts-se-whereby) [inline src/utils/types.ts:1]: At least some of the types are defined but have no references.
We put them to use in next iteration?

File: src/utils/types.ts (hunk 2 above)
Comment (thyal) [reply src/utils/types.ts:17]: nice catch, will fix

File: src/utils/types.ts
Comment (thyal) [reply src/utils/types.ts:1]: Most of these are moved from the browser-sdk repo, and all of them are used there (we export the types here). They should all be used in the correct places in this repo, but that will be an iteration, I think. For now we just export them and use them in the SDK. See this PR: https://github.com/whereby/sdk/pull/112

File: src/webrtc/P2pRtcManager.ts [hunk 3]
```diff
@@ -959,7 +1019,7 @@ export default class P2pRtcManager {
 
     // implements a strategy to change the bandwidth for all clients (without negotiation)
     // returns bandwidth so it can be used as initial bandwidth for new client.
-    _changeBandwidthForAllClients(isJoining) {
+    _changeBandwidthForAllClients(isJoining: any) {
```
Comment (pnts-se-whereby) [inline src/webrtc/P2pRtcManager.ts:1022]: I think this is a boolean

File: src/webrtc/VegaConnection.ts [hunk 4]
```diff
@@ -25,6 +30,8 @@ export default class VegaConnection extends EventEmitter {
     }
 
     _tearDown() {
+        if (this.socket === null) return;
```
Comment (pnts-se-whereby) [inline src/webrtc/VegaConnection.ts:33]: I think we still want to do the following, even though we don't have a socket:

```
       this.sents.forEach((sent) => sent.close());

        this.emit("close");
```

File: .eslintrc (hunk 1 above)
Comment (sebapawlak) [reply .eslintrc:17]: Let's add this back

Comment (thyal) [reply .eslintrc:17]: done 👍
